enterprise/api_client/discovery.py

```diff
@@ -132,11 +144,22 @@ def get_catalog_results(self, content_filter_query, query_params=None, traverse_
         query_params = query_params or {}
 
         try:
-            endpoint = getattr(self.client, self.SEARCH_ALL_ENDPOINT)
-            response = endpoint().post(data=content_filter_query, **query_params)
-            if traverse_pagination:
-                response['results'] = self.traverse_pagination(response, endpoint, content_filter_query, query_params)
-                response['next'] = response['previous'] = None
+            cache_key = utils.get_cache_key(
+                service='discovery',
+                endpoint=self.SEARCH_ALL_ENDPOINT,
+                query=content_filter_query,
+                traverse_pagination=traverse_pagination,
+                **query_params
+            )
+            response = cache.get(cache_key)
+            if not response:
+                # Response is not cached, so make a call.
+                response = self._get_catalog_results_from_discovery(
+                    content_filter_query,
+                    query_params,
+                    traverse_pagination
+                )
+                cache.set(cache_key, response, settings.DISCOVERY_GET_CATALOG_RESULT_CACHE_TIMEOUT)
```

Comment:
instead of creating `DISCOVERY_GET_CATALOG_RESULT_CACHE_TIMEOUT ` in this we should use `ENTERPRISE_API_CACHE_TIMEOUT`.

Reply:
I think ENTERPRISE_API_CACHE_TIMEOUT is used for API that belongs to Enterprise. That is why we are using it in EnterpriseApiClient.
But now we want some TimeOut period for Discovery to use in CourseCatalogApiClient.

Reply:
That time out `ENTERPRISE_API_CACHE_TIMEOUT ` is used in cache results of discovery call. since that enterprise API is calling discovery.